tests/test_conversion.py

```diff
@@ -84,3 +84,13 @@ def test_bits_upcast(T):
 def test_bit_upcast():
     x = m.Enable()
     assert type(m.bit(x)) is m.Bit
+
+
+def test_name():
```

Comment:
can we parameterize this test by a handful of conversions? and additionally check that the conversion happened properly?